### ISSUE TITLE
Feature request: add auto expiration duration

### ISSUE THREAD

**Issue by lefuturiste** [sentriz/cliphist#100]
Hi!
I think it's pretty useful to have a way to clear entries older that 1 hour every now and then.

**Comment by rogervn** [comment]
Is there a reason why this hasn't merged? It's such an useful feature.

**Comment by zygimantus** [comment]
Will this be merged?

**Comment by luisnquin** [comment]
Nah, the pull request was closed because the author wants to **keep things simple**.

It is a pretty useful feature and might even become important in my workflow, I'm considering forking this project to allow it.

**Comment by DanielKaleby** [comment]
I was already bitten by this penguin, but I let it go when I saw that it is a layer above the cliphist.
There are several options, like a Cronjob reading the file and cleaning it.
Small and scripted code> try to predict each possible use case.
In the command line it turns out that in the end the less is the more, and the more focused a tool is, the more adaptable it becomes.

Instead of deleting old entrances, I prefer to encrypt the history, clean manually periodically, just like in Bash.

**Comment by starblursd** [comment]
my solution for things like passwords is `cliphist list | head -n1 | cut -f1 | cliphist delete` pipes output of list into parsing just the first entry (most recent), grabs the id by cutting up to the tab, then pipes that into delete to remove that entry.

and then set that on a keybind (i use niri so its a `spawn-sh "cliphist list | head -n1 | cut -f1 | cliphist delete"`)

**Comment by coderkun** [comment]
Looks like cliphist already [respects the “sensitive” flag](https://github.com/sentriz/cliphist/blob/f49bd905cff72d32d62c209224353865436f9a13/cliphist.go#L69). In that case you only need to make sure passwords are added to the clipboard with `CLIPBOARD_STATE=sensitive`, e. g. via `wl-copy --sensitive`.

**Comment by sentriz** [comment]
added `cliphist wipe -older-than <time>` 👍

**Comment by varac** [comment]
@sentriz Great, thanks ! Could you cut a release including this new feature pls ?
